File: app/router/auth_router.py
from fastapi import APIRouter
from app.schemas.auth_schema import LoginSchema, TokenResponse, RegistroSchema
from app.controllers.auth_controller import AuthController

router = APIRouter(prefix="/auth", tags=["Autenticación"])

@router.post("/login", response_model=TokenResponse)
def login(credenciales: LoginSchema):
    return AuthController.login(credenciales)

@router.post("/registro", status_code=201)
def registro(datos: RegistroSchema):
    return AuthController.registro(datos)

File: src/services-api/auth_router.js
import { fastApi } from '../api/fastApi';

api = fastApi;

export const AuthService = {
    // Iniciar sesión (POST /auth/login)
    login: async (credenciales) => {
        const response = await api.post('/auth/login', credenciales);
        return response.data;
    },

    registro: async (datosUsuario) => {
        const response = await api.post('/auth/registro', datosUsuario);
        return response.data;
    },

    logout: async () => {
        const response = await api.post('/auth/logout');
        return response.data;
    }
};